packages/eas-cli/src/build/android/UpdatesModule.ts

```diff
@@ -3,12 +3,13 @@ import { AndroidConfig } from '@expo/config-plugins';
 import fs from 'fs-extra';
 
 import log from '../../log';
+import { getProjectAccountName } from '../../project/projectUtils';
 import { ensureLoggedInAsync } from '../../user/actions';
 import { ensureValidVersions } from '../utils/updates';
 
 export async function configureUpdatesAsync(projectDir: string, exp: ExpoConfig): Promise<void> {
   ensureValidVersions(exp);
-  const { username } = await ensureLoggedInAsync();
+  const accountName = getProjectAccountName(exp, await ensureLoggedInAsync());
```

Comment:
Since the user account from `ensureLoggedInAsync` is only used if the `owner` field is not specified, I wonder if it would be worth it to have a variant of `getProjectAccountName` (or `fetchProjectAccountNameAsync`) that would only fetch the user account if `owner` is not specified? (As an optimization.)

Reply:
Yeah, that's a good one. We can definitely add that!

Reply:
Also, only in the `UpdatesModule` scenario, the user doesn't have to be authenticated. In all other scenarios, such as creating releases, the user _has_ to be authenticated regardless of the owner. We can still do it, but the eventual improvements are limited.

Reply:
Having to be logged in is different from having to fetch the user profile from the server, because the username is stored locally in the session file.

I think ideally we wouldn't fetch a user profile from the server unless we have to.

In Expo CLI we would use [`getCurrentUsernameAsync`](https://github.com/expo/expo-cli/blob/d60df2a1f7980b5c47c2c64bee825d5071ac077f/packages/xdl/src/User.ts#L289), which doesn't make an HTTP  request if you're a person, and if you're a robot, well in that case `owner` is mandatory, so we still wouldn't fetch the user profile.